opnode/rollup/driver/state.go

```diff
@@ -282,6 +313,39 @@ func (s *state) handleEpoch(ctx context.Context) (bool, error) {
 
 }
 
+func (s *state) handleUnsafeL2Payload(ctx context.Context, payload *l2.ExecutionPayload) error {
+	if s.l2SafeHead.Number > uint64(payload.BlockNumber) {
+		s.log.Info("ignoring unsafe L2 execution payload, already have safe payload", "id", payload.ID())
+		return nil
+	}
+
+	if uint64(payload.Timestamp) < s.l2SafeHead.Time {
+		s.log.Warn("received payload rewinds L2 chain beyond current L2 safe head")
+	}
+
+	if uint64(payload.Timestamp) > s.l2Head.Time+s.Config.BlockTime {
```

Comment:
Is the intention to only advance on block time at a time? But this does allow reorgs, right?

Reply:
We should allow reorgs, but not too far in the future. This is not consensus code, so a heuristic to not try and add blocks beyond the expected timestamp seemed ok. I can remove it if you prefer that blocks received on p2p can trigger more.

Reply:
Thought about this more and think it's fine to remove the heuristics here, as the execution engine will handle them anyway, and the heuristics were not that nice. Removed in commit [a069290](https://github.com/ethereum-optimism/optimistic-specs/pull/394/commits/a069290a999cea7290ce35aad32d3fd200cffcd6), and added a comment about reorg/sync behavior.